src/jit/gtlist.h

```diff
@@ -155,6 +155,7 @@ GTNODE(GT               , ">"            ,GenTreeOp          ,0,GTK_BINOP|GTK_RE
 // Because of this there is no need to also add GT_TEST_LT/LE/GE/GT opers.
 GTNODE(TEST_EQ          , "testEQ"       ,GenTreeOp          ,0,GTK_BINOP|GTK_RELOP)
 GTNODE(TEST_NE          , "testNE"       ,GenTreeOp          ,0,GTK_BINOP|GTK_RELOP)
+GTNODE(CMP              , "cmp"          ,GenTreeOp          ,0,GTK_BINOP|GTK_NOVALUE)
```

Comment:
My main concern with making `cmp` a relop is that some code may try to reverse it but that's not possible since there's no condition associated with it. There's also the fact that unlike other relops `cmp` does not return a value and thus it needs some special casing (e.g. `dstCount` setting in `TreeNodeInfoInitCmp`).

We do end up with some code like `else if (tree->OperIsCompare() || tree->OperIs(GT_CMP))` that perhaps looks a bit strange. As stated in the initial PR comment it would be better to lower all relops to avoid this kind of duplication. See #10861 for how that may look.